src/sparse_blas/backends/mkl_common/mkl_operations.cxx

```diff
@@ -0,0 +1,128 @@
+}
+
+template <typename fpType>
+std::enable_if_t<detail::is_fp_supported_v<fpType>> gemv(
+    sycl::queue& queue, transpose transpose_val, const fpType alpha,
+    detail::matrix_handle* A_handle, sycl::buffer<fpType, 1>& x, const fpType beta,
+    sycl::buffer<fpType, 1>& y) {
+    oneapi::mkl::sparse::gemv(queue, transpose_val, alpha, detail::get_handle(A_handle), x, beta, y);
+}
+
+template <typename fpType>
+std::enable_if_t<detail::is_fp_supported_v<fpType>, sycl::event> gemv(
+    sycl::queue& queue, transpose transpose_val, const fpType alpha,
+    detail::matrix_handle* A_handle, const fpType* x, const fpType beta, fpType* y,
+    const std::vector<sycl::event>& dependencies) {
+    return oneapi::mkl::sparse::gemv(queue, transpose_val, alpha, detail::get_handle(A_handle), x, beta, y,
+                                     dependencies);
+}
+
+template <typename fpType>
+std::enable_if_t<detail::is_fp_supported_v<fpType>> trsv(sycl::queue& /*queue*/, uplo /*uplo_val*/,
+                                                         transpose /*transpose_val*/,
+                                                         diag /*diag_val*/,
+                                                         detail::matrix_handle* /*A_handle*/,
+                                                         sycl::buffer<fpType, 1>& /*x*/,
+                                                         sycl::buffer<fpType, 1>& /*y*/) {
+    throw unimplemented("SPARSE_BLAS", "trsv");
+}
+
+template <typename fpType>
+std::enable_if_t<detail::is_fp_supported_v<fpType>, sycl::event> trsv(
+    sycl::queue& /*queue*/, uplo /*uplo_val*/, transpose /*transpose_val*/, diag /*diag_val*/,
+    detail::matrix_handle* /*A_handle*/, const fpType* /*x*/, fpType* /*y*/,
+    const std::vector<sycl::event>& /*dependencies*/) {
+    throw unimplemented("SPARSE_BLAS", "trsv");
+}
+
+template <typename fpType>
+std::enable_if_t<detail::is_fp_supported_v<fpType>> gemm(
+    sycl::queue& queue, layout dense_matrix_layout, transpose transpose_A, transpose transpose_B,
+    const fpType alpha, detail::matrix_handle* A_handle, sycl::buffer<fpType, 1>& B,
+    const std::int64_t columns, const std::int64_t ldb, const fpType beta,
+    sycl::buffer<fpType, 1>& C, const std::int64_t ldc) {
+    oneapi::mkl::sparse::gemm(queue, dense_matrix_layout, transpose_A, transpose_B, alpha,
+                              detail::get_handle(A_handle), B, columns, ldb, beta, C, ldc);
+}
+
+template <typename fpType>
+std::enable_if_t<detail::is_fp_supported_v<fpType>, sycl::event> gemm(
+    sycl::queue& queue, layout dense_matrix_layout, transpose transpose_A, transpose transpose_B,
+    const fpType alpha, detail::matrix_handle* A_handle, const fpType* B,
+    const std::int64_t columns, const std::int64_t ldb, const fpType beta, fpType* C,
+    const std::int64_t ldc, const std::vector<sycl::event>& dependencies) {
+    // TODO: Remove const_cast with 2024.0 oneMKL release
```

Comment:
@gajanan-choudhary even with the latest MKL build this `const_cast` cannot be removed yet. Are you planning to make the `B` argument const? This may be too late for the 2024.0 release right?

Reply:
Similar question for `optimize_gemm` which does not seem to be implemented yet.